src/univariate/discrete/betabinomial.jl

```diff
@@ -82,18 +82,22 @@ end
 
 function pdf(d::BetaBinomial, k::Int)
     n, α, β = d.n, d.α, d.β
-    choose = 1 / ((n+1) * beta(k+1, n-k+1))
+    chooseinv = (n+1) * beta(k+1, n-k+1)
     numerator = beta(k + α, n - k + β)
     denominator = beta(α, β)
-    return choose * (numerator / denominator)
+    return (numerator / denominator) / chooseinv
 end
 
 function pdf(d::BetaBinomial)
     n, α, β = d.n, d.α, d.β
-    k = 0:n
-    binoms = [1 / ((n+1) * beta(i+1, n-i+1)) for i in k]
-    fixed_beta = beta(α, β)
-    return binoms .* @compat(beta.(k + α, n - k + β)) / fixed_beta
+    denominator = beta(α, β)
+    values = Array{Float64, 1}()
+    for k = 0:n
+        chooseinv = (n+1) * beta(k+1, n-k+1)
+        numerator = beta(k + α, n - k + β)
+        push!(values, (numerator / denominator) / chooseinv)
```

Comment:
> `push!` doesn't appear to have convert functionality, could you explain the conversion concern?

It's not that `push!` itself converts, it's that the array is defined to have element type `Float64`, so Julia will _attempt_ to convert any value that's added to the array, be it via `push!`, `setindex!`, or what have you, to the array's defined element type. That's evident in the example you gave:
```julia
julia> push!(a, "A")
ERROR: MethodError: Cannot `convert` an object of type String to an object of type Float64
This may have arisen from a call to the constructor Float64(...),
since type constructors fall back to convert methods.
 in push!(::Array{Float64,1}, ::String) at ./array.jl:479
```
It's trying to convert the value you gave it, but in this case the conversion fails. If you were to push a `Float32` or `Int64`, for example, or any type for which a `convert` method exists for `Float64`, the conversion wouldn't fail. However, then you're locked into `Float64`; the natural type that would have resulted from `promote_op` is no longer preserved.

Do you see what I mean?